pkg/cri/server/image_pull.go

```diff
@@ -126,6 +129,17 @@ func (c *criService) PullImage(ctx context.Context, r *runtime.PullImageRequest)
 		}
 	)
 
+	if vAddr := c.config.ImageVerification.Address; vAddr != "" {
+		ctx, cancel := context.WithTimeout(ctx, 1*time.Second)
+		defer cancel()
+		conn, err := dialer.ContextDialer(ctx, "unix://"+vAddr)
+		if err != nil {
+			return nil, fmt.Errorf("failed to connect to image verification plugin: %w", err)
+		}
+		client := image_verifier.NewImageVerifierClient(ttrpc.NewClient(conn))
+		resolver = NewVerifyingResolver(resolver, client)
```

Comment:
Perhaps this should be implemented as a `client.RemoteOpt` rather than a resolver?

Because potentially there can be a code path to skip calling the resolver (in a future version of containerd).